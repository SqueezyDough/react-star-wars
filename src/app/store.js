import { createStore } from 'redux'
import { persistStore, persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage'
import resourceReducer from '../features/resource/resourceSlice'

const persistConfig = {
  key: 'root',
  storage
}

const persistedReducer = persistReducer(persistConfig, resourceReducer)

export const store = createStore(
  persistedReducer,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
)
export const persistor = persistStore(store)
