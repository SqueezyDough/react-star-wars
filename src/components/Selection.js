import React from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { changeValue } from '../features/resource/resourceSlice'
import styled from 'styled-components'

function Select ({ className, options }) {
  const { selectedOption } = useSelector(state => state)
  const dispatch = useDispatch()

  const handleChange = e => {
    dispatch(changeValue(e.target.value))
  }

  return (
    <div className={className}>
      <select onChange={handleChange} value={selectedOption}>
        {options.map(option => (
          <option value={option.value.name} key={option.value.name}>{option.value.name}</option>
        ))}
      </select>
    </div>
  )
}

export default styled(Select)`
  display: inline;

  select {
    margin: 3rem .5rem 0;
    padding: .5rem 1rem;
    border: none;
    font-size: ${({ theme }) => theme.fontSizeXS}
  }
`
