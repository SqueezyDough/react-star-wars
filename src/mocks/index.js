export const characterWithTitle = {
  title: 'title',
  url: 'http://swapi.dev/api/people/1/',
  props: {
    mass: 77,
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19bby',
    gender: 'male',
    homeworld: 'http://swapi.dev/api/planets/1/'
  }
}

export const characterWithName = {
  name: 'name',
  url: 'http://swapi.dev/api/people/1/',
  props: [{
    mass: 77,
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19bby',
    gender: 'male',
    homeworld: 'http://swapi.dev/api/planets/1/'
  }]
}

export const option = {
  value: {
    name: 'name',
    url: 'url'
  }
}

export const options = [option]

export const resource = {
  name: 'people'
}

export const loadingState = {
  start: true,
  loaders: 8
}
