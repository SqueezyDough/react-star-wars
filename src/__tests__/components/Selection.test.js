import React from 'react'
import renderer from 'react-test-renderer'
import Selection from '../../components/Selection'
import { option, options } from '../../mocks/index'
import { Provider } from 'react-redux'
import { store, persistor } from '../../app/store'
import { PersistGate } from 'redux-persist/integration/react'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <PersistGate loading={null} persistor={persistor}>
          <Selection option={option} options={options} />
        </PersistGate>
      </Provider>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
