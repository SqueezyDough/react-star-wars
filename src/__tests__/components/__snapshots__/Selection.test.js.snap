// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  display: inline;
}

.c0 select {
  margin: 3rem .5rem 0;
  padding: .5rem 1rem;
  border: none;
}

<div
  className="c0"
>
  <select
    onChange={[Function]}
    value=""
  >
    <option
      value="name"
    >
      name
    </option>
  </select>
</div>
`;
